File: src/main/java/org/xmolecules/ide/intellij/JPA.java
package org.xmolecules.ide.intellij;

import com.intellij.openapi.roots.ExternalLibraryDescriptor;
import nexos.intellij.ddd.Framework;

import java.util.List;

public class JPA {
    public static final Framework JPA = new Framework("JPA");
    public static final Library LIB = new Library(JPA, new ExternalLibraryDescriptor("javax.persistence", "javax.persistence-api"));

    public static final ConceptImplementation Entity =  new ConceptViaTypeAnnotation(Concepts.ENTITY, LIB, "javax.persistence.Entity");
    public static final ConceptImplementation Embeddable =  new ConceptViaTypeAnnotation(Concepts.VALUE_OBJECT, LIB, "javax.persistence.Embeddable");
    public static final List<ConceptImplementation> ALL = List.of(Entity, Embeddable);
}

File: src/main/kotlin/nexos/intellij/ddd/jpa.kt
package nexos.intellij.ddd.jpa

import com.intellij.openapi.roots.ExternalLibraryDescriptor
import nexos.intellij.ddd.Framework
import nexos.intellij.ddd.Info
import nexos.intellij.ddd.Library
import nexos.intellij.ddd.Entity as ddd_Entity
import nexos.intellij.ddd.ValueObject as ddd_ValueObject

object JPA: Framework("JPA") {
    //Version 2.2, 2.3-SNAPSHOT
    val all by lazy { listOf(Entity, Embeddable) }
}
val lib = Library(JPA, ExternalLibraryDescriptor("javax.persistence", "javax.persistence-api"))

val Entity =  Info("javax.persistence.Entity", ddd_Entity, lib)
val Embeddable =  Info("javax.persistence.Embeddable", ddd_ValueObject, lib)